### PR TITLE
Adds 'TrivialMapElimination' and 'TrivialMapRangeElimination' and tests.

### PR DESCRIPTION
**Pull request by PanicSheep**

### CODE REVIEW

**Comment by PanicSheep** [comment]
Discussed with @am-ivanov to add tests to the transformations before merging.

**Review by tbennun — CHANGES_REQUESTED** [review]
small but fundamental change is necessary

**Comment by tbennun** [comment]
@PanicSheep I would approve this PR if there was a test for it

File: dace/transformation/dataflow/trivial_map_elimination.py
Comment (PanicSheep) [reply dace/transformation/dataflow/trivial_map_elimination.py]: See commit "Only replaces map parameter on subgraph."

File: dace/transformation/dataflow/trivial_map_range_elimination.py
Comment (PanicSheep) [reply dace/transformation/dataflow/trivial_map_range_elimination.py]: See commit "Only replaces map parameter on subgraph."

File: dace/transformation/dataflow/trivial_map_elimination.py
Comment (PanicSheep) [reply dace/transformation/dataflow/trivial_map_elimination.py]: See commit "Only replaces map parameter on subgraph."

File: tests/trivial_map_elimination.py
Comment (tbennun) [inline tests/trivial_map_elimination.py:1]: please add a `_test` suffix to the file so that it gets run automatically in CI. (we check for `tests/*_test.py`)

File: tests/trivial_map_range_elimination_test.py
Comment (tbennun) [inline tests/trivial_map_range_elimination_test.py:16]: This is an illegal map. the values of `k` cannot depend on `j`

Comment (tbennun) [reply tests/trivial_map_range_elimination_test.py:16]: weird that it passes validation...

**Review by tbennun — CHANGES_REQUESTED** [review]
Almost done, rename a file and fix a range and it's approved!

**Review by tbennun — APPROVED** [review]

**Pull request merged**
